MediaBrowser.Common/Providers/ProviderIdParsers.cs

```diff
@@ -0,0 +1,125 @@
+
+                text = text.Slice(ttPos + tt.Length);
+                var i = 0;
+                for (; i < Math.Min(text.Length, ImdbMaxNumbers); i++)
+                {
+                    var c = text[i];
+                    if (!IsDigit(c))
+                    {
+                        break;
+                    }
+                }
+
+                // skip if more than 8 digits
+                if (i <= ImdbMaxNumbers && i >= ImdbMinNumbers)
+                {
+                    imdbId = string.Concat(tt, text.Slice(0, i));
+                    return true;
+                }
+
+                text = text.Slice(i);
+            }
+
+            imdbId = default;
+            return false;
+        }
+
+        /// <summary>
+        /// Parses an TMDb id from a movie url.
+        /// </summary>
+        /// <param name="text">The text with the url to parse.</param>
+        /// <param name="tmdbId">The parsed TMDb id.</param>
+        /// <returns>True if parsing was successful, false otherwise.</returns>
+        public static bool TryParseTmdbMovieId(ReadOnlySpan<char> text, [NotNullWhen(true)] out string? tmdbId)
+            => TryParseProviderId(text, "themoviedb.org/movie/", out tmdbId);
+
+        /// <summary>
+        /// Parses an TMDb id from a series url.
+        /// </summary>
+        /// <param name="text">The text with the url to parse.</param>
+        /// <param name="tmdbId">The parsed TMDb id.</param>
+        /// <returns>True if parsing was successful, false otherwise.</returns>
+        public static bool TryParseTmdbSeriesId(ReadOnlySpan<char> text, [NotNullWhen(true)] out string? tmdbId)
+            => TryParseProviderId(text, "themoviedb.org/tv/", out tmdbId);
+
+        /// <summary>
+        /// Parses an TVDb id from a url.
+        /// </summary>
+        /// <param name="text">The text with the url to parse.</param>
+        /// <param name="tvdbId">The parsed TVDb id.</param>
+        /// <returns>True if parsing was successful, false otherwise.</returns>
+        public static bool TryParseTvdbId(ReadOnlySpan<char> text, [NotNullWhen(true)] out string? tvdbId)
+            => TryParseProviderId(text, "thetvdb.com/?tab=series&id=", out tvdbId);
+
+        private static bool TryParseProviderId(ReadOnlySpan<char> text, ReadOnlySpan<char> searchString, [NotNullWhen(true)] out string? providerId)
+        {
+            var searchPos = text.IndexOf(searchString);
+            if (searchPos == -1)
+            {
+                providerId = default;
+                return false;
+            }
+
+            text = text.Slice(searchPos + searchString.Length);
+
+            int i = 0;
+            for (; i < text.Length; i++)
+            {
+                var c = text[i];
+
+                if (!IsDigit(c))
+                {
+                    break;
+                }
+            }
+
+            if (i >= 1)
+            {
+                providerId = text.Slice(0, i).ToString();
```

Comment:
Don't `ToString`, the caller can do that when needed, that way we don't allocate if we jus want to know if it contains an id